src/Controller/UploadedFileController.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace EMS\CoreBundle\Controller;
+
+use EMS\CoreBundle\Entity\UploadedAsset;
+use EMS\CoreBundle\Form\Data\EntityTable;
+use EMS\CoreBundle\Form\Data\TableAbstract;
+use EMS\CoreBundle\Form\Form\TableType;
+use EMS\CoreBundle\Service\FileService;
+use Psr\Log\LoggerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Form\Form;
+use Symfony\Component\Form\SubmitButton;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class UploadedFileController extends AbstractController
+{
+    /** @var string */
+    public const SOFT_DELETE_ACTION = 'soft_delete';
+    /** @var string */
+    public const HARD_DELETE_ACTION = 'hard_delete';
+
+    private LoggerInterface $logger;
+    private FileService $fileService;
+
+    public function __construct(LoggerInterface $logger, FileService $fileService)
+    {
+        $this->logger = $logger;
+        $this->fileService = $fileService;
+    }
+
+    public function index(Request $request): Response
+    {
+        $table = new EntityTable($this->fileService);
+        $column = $table->addColumn('uploaded-file.index.column.name', 'name');
+        $column->setRoutePath('ems_file_download', function (UploadedAsset $data) {
+            return [
+                'sha1' => $data->getSha1(),
+                'type' => $data->getType(),
+                'name' => $data->getName(),
+            ];
+        });
+        $column->setRouteTarget('_blank');
+        $table->addColumn('uploaded-file.index.column.sha1', 'sha1');
+        $table->addColumn('uploaded-file.index.column.type', 'type');
+        $table->addColumn('uploaded-file.index.column.username', 'user');
+        $tableColumn = $table->addColumn('uploaded-file.index.column.created', 'created');
+        $tableColumn->setDateTimeProperty(true);
+        $tableColumn = $table->addColumn('uploaded-file.index.column.size', 'size');
+        $tableColumn->setFormatBytes(true);
```

Comment:
```suggestion
        $table
            ->addColumn('uploaded-file.index.column.name', 'name')
            ->setRoutePath('ems_file_download', function (UploadedAsset $data) {
                return [
                    'sha1' => $data->getSha1(),
                    'type' => $data->getType(),
                    'name' => $data->getName(),
                ];
            })
            ->setRouteTarget('_blank');
        $table->addColumn('uploaded-file.index.column.sha1', 'sha1');
        $table->addColumn('uploaded-file.index.column.type', 'type');
        $table->addColumn('uploaded-file.index.column.username', 'user');
        $table
            ->addColumn('uploaded-file.index.column.created', 'created')
            ->setDateTimeProperty(true);
        $table
            ->addColumn('uploaded-file.index.column.size', 'size')
            ->setFormatBytes(true);
```

Reply:
let me know what you think about this strategy @Davidmattei and @theus77

Reply:
I have stated another PR in order to simplify this. For instance with Column classes by type of column such as UserTableColumn, DatetimeTableColumn, BytesTableColumn

Reply:
Perfect!